pkg/azureutils/azure_disk_utils.go

```diff
@@ -0,0 +1,288 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azureutils
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"strings"
+	"unicode"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-12-01/compute"
+	"github.com/pborman/uuid"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/volume/util"
+
+	azDiskClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+	azure "sigs.k8s.io/cloud-provider-azure/pkg/provider"
+)
+
+const (
+	azureStackCloud = "AZURESTACKCLOUD"
+
+	azurePublicCloudDefaultStorageAccountType = compute.StandardSSDLRS
+	azureStackCloudDefaultStorageAccountType  = compute.StandardLRS
+	defaultAzureDataDiskCachingMode           = v1.AzureDataDiskCachingReadOnly
+
+	// default IOPS Caps & Throughput Cap (MBps) per https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disks-ultra-ssd
+	// see https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
+	diskNameMinLength = 1
+	diskNameMaxLength = 80
+	// maxLength = 80 - (4 for ".vhd") = 76
+	diskNameGenerateMaxLength = 76
+
+	// minimum disk size is 1GiB
+	MinimumDiskSizeGiB = 1
+
+	// DefaultAzureCredentialFileEnv is the default azure credentials file env variable
+	DefaultAzureCredentialFileEnv = "AZURE_CREDENTIAL_FILE"
+	// DefaultCredFilePathLinux is default creds file for linux machine
+	DefaultCredFilePathLinux = "/etc/kubernetes/azure.json"
+	// DefaultCredFilePathWindows is default creds file for windows machine
+	DefaultCredFilePathWindows = "C:\\k\\azure.json"
+
+	// String constants used by CloudProvisioner
+	ResourceNotFound         = "ResourceNotFound"
+	ErrDiskNotFound          = "not found"
+	SourceSnapshot           = "snapshot"
+	SourceVolume             = "volume"
+	DriverName               = "disk.csi.azure.com"
+	CSIDriverMetricPrefix    = "azuredisk_csi_driver"
+	CreatedForPVNameKey      = "kubernetes.io-created-for-pv-name"
+	ResizeRequired           = "resizeRequired"
+	SourceDiskSearchMaxDepth = 10
+	CachingModeField         = "cachingmode"
+	StorageAccountTypeField  = "storageaccounttype"
+	StorageAccountField      = "storageaccount"
+	SkuNameField             = "skuname"
+	LocationField            = "location"
+	ResourceGroupField       = "resourcegroup"
+	DiskIOPSReadWriteField   = "diskiopsreadwrite"
+	DiskMBPSReadWriteField   = "diskmbpsreadwrite"
+	DiskNameField            = "diskname"
+	DesIDField               = "diskencryptionsetid"
+	TagsField                = "tags"
+	MaxSharesField           = "maxshares"
+	IncrementalField         = "incremental"
+	LogicalSectorSizeField   = "logicalsectorsize"
```

Comment:
I spot checked a few of these constants and all were unused. Are these necessary?

Reply:
These constants are used in cloudprovisioner_azuredisk.go file. These constants are keys in the parameters of the request which are specific to azure as of now.